### DISCUSSION TITLE
Reduced RAM usage

### DISCUSSION THREAD

**Discussion by bradbeattie** [healthchecks/healthchecks#907] [Show and tell]
For self-hosted Prometheus-only setups, the following dramatically reduced the memory footprint of my Healthchecks instance:

```docker
FROM healthchecks/healthchecks:latest
USER root
RUN sed -i 's/processes = .*/processes = 1/g' /opt/healthchecks/docker/uwsgi.ini
RUN sed -i 's/attach-daemon =.*//g' /opt/healthchecks/docker/uwsgi.ini
USER hc
```

**Comment by cuu508** [comment]
uWSGI supports configuration via environment variables: https://uwsgi-docs.readthedocs.io/en/latest/Configuration.html#environment-variables

But looks like if an option is set both in a configuration file, and in the environment variable then the configuration file takes precedence.

[I updated uwsgi.ini](https://github.com/healthchecks/healthchecks/commit/89aa0a832b18f52dff3051b9c872f7a5d6ced807) to allow the `UWSGI_PROCESSES` environment to override the default value (4) set in uwsgi.ini. So with future releases you will be able to set `UWSGI_PROCESSES` in the .env file.

**Reply by bradbeattie** [reply]
Thanks. That was one of my discoveries as well which necessitated the sed approach. Disabling the additional daemons made a significant impact too, but I don't anticipate an envvar patch for that one. ;)
